### PR TITLE
[ci skip] Enhanced rdoc for Hash.new and Hash.try_convert

### PR DESCRIPTION
**Pull request by BurdetteLamar**
This PR changes only rdoc comments.

### CODE REVIEW

**Comment by esparta** [comment]
I took the liberty of use rdoc to render the before after of the changes:
`new`
(before)

image

(after)
image

`try_convert`
(before)
image

(after)
image

File: hash.c [hunk 1]
```diff
@@ -1745,38 +1745,48 @@ set_proc_default(VALUE hash, VALUE proc)
...
+ *
+ *  If argument <tt>default_value</tt> given but no block given,
+ *  initializes the default value to the given <tt>default_value</tt>
+ *  and the default proc to <tt>nil</tt>:
+ *
+ *    h = Hash.new(false)
+ *    h # => {}
+ *    h.default # => false
+ *    h.default_proc # => nil
+ *    h[:nosuch] # => nil
```
Comment (esparta) [inline hash.c]: This is not correct:
```ruby
h = Hash.new(false)
# => {}
h[:nosuch]
# => false
```

Comment (BurdetteLamar) [reply hash.c]: Thanks, esparta, for this careful review.  Now fixed.

**Comment by BurdetteLamar** [comment]
Will try new PR.

**Pull request closed without merging**